src/components/molecules/PopapPokemons/PopapPokemons.tsx

```diff
@@ -0,0 +1,47 @@
+import { useFetch } from "../../../hooks/useFetch";
+import { ReturnResult } from "../../../models/Pokemons"
+import { ResponsePokemon } from "../../../models/ResponsePokemon";
+
+interface Props {
+  pokemon: ReturnResult;
+  setShowPopup: () => void;
+}
+
+
+export const PopapPokemons = ({pokemon, setShowPopup}:Props) => {
+
+ const {data, isLoading} = useFetch<ResponsePokemon>(`https://pokeapi.co/api/v2/pokemon/${pokemon.id}`)
+
+ const getAbilities = data?.abilities.map((item)=> ({
+  ability:item.ability.name
+ }))
+ const getMoves = data?.moves.map((item)=>({
+  moves:item.move.name
+ }) ).filter((pokemon, index) =>  index < 3);
```

Comment:
I don't think you need to map and extract each ability and move from data, you can directly map the Pokemon's abilities and moves in the component